tests/CPT_test.py

```diff
@@ -0,0 +1,332 @@
+            label_tokenized = self._get_input_ids(label)
+            label_template_part2_tokenized = self._get_input_ids(label_template_part_2)
+
+            eos = [self.tokenizer.eos_token_id] if self.tokenizer.eos_token_id is not None else []
+            input_ids = (
+                input_template_tokenized_part1
+                + input_tokenized
+                + input_template_tokenized_part2
+                + sep_tokenized
+                + label_template_part1_tokenized
+                + label_tokenized
+                + label_template_part2_tokenized
+                + eos
+            )
+
+            # determine label tokens, to calculate loss only over them when labels_loss == True
+            attention_mask = [1] * len(input_ids)
+            input_type_mask = (
+                [1] * len(input_template_tokenized_part1)
+                + [2] * len(input_tokenized)
+                + [1] * len(input_template_tokenized_part2)
+                + [0] * len(sep_tokenized)
+                + [3] * len(label_template_part1_tokenized)
+                + [4] * len(label_tokenized)
+                + [3] * len(label_template_part2_tokenized)
+                + [0] * len(eos)
+            )
+
+            assert len(input_type_mask) == len(input_ids) == len(attention_mask)
+
+            return input_ids, attention_mask, input_type_mask
+
+        def __len__(self):
+            return len(self.input_ids)
+
+        def __getitem__(self, idx):
+            return {
+                "input_ids": self.input_ids[idx],
+                "attention_mask": self.attention_mask[idx],
+                "input_type_mask": self.input_type_mask[idx],
+            }
+
+    dataset = CPTDataset(data, tokenizer, TEMPLATE)
+
+    return dataset
+
+
+def test_model_initialization_text(global_tokenizer, config_text):
+    """Test model loading and PEFT model initialization."""
+    base_model = AutoModelForCausalLM.from_pretrained(MODEL_NAME, cache_dir=".", trust_remote_code=True)
+
+    model = get_peft_model(base_model, config_text)
+    assert model is not None, "PEFT model initialization failed"
+
+
+def test_model_initialization_random(global_tokenizer, config_random):
+    """Test model loading and PEFT model initialization."""
+    base_model = AutoModelForCausalLM.from_pretrained(MODEL_NAME, cache_dir=".", trust_remote_code=True)
+
+    model = get_peft_model(base_model, config_random)
+    assert model is not None, "PEFT model initialization failed"
+
+
+def test_model_training_random(sst_data, global_tokenizer, collator, config_random):
+    """Perform a short training run to verify the model and data integration."""
+
+    base_model = AutoModelForCausalLM.from_pretrained(MODEL_NAME, cache_dir=".", trust_remote_code=True)
+    model = get_peft_model(base_model, config_random)
+    emb = model.prompt_encoder.default.embedding.weight.data.clone().detach()
+    training_args = TrainingArguments(
+        output_dir="./results",
```

Comment:
Let's use a temporary directory here. You can use the `tmp_path` fixture from pytest.